spot-ml/src/main/scala/org/apache/spot/SpotSparkLDAWrapper.scala

```diff
@@ -34,43 +35,59 @@ object SpotSparkLDAWrapper {
              ldaBeta: Double = 1.001,
              maxIterations: Int = 20): SpotLDAOutput = {
 
+    import sqlContext.implicits._
+
+    val docWordCountCache = docWordCount.cache()
+
+    // Forcing an action to cache results.
+    docWordCountCache.count()
+
     // Create word Map Word,Index for further usage
     val wordDictionary: Map[String, Int] = {
-      val words = docWordCount
-        .cache
+      val words = docWordCountCache
         .map({ case SpotLDAInput(doc, word, count) => word })
         .distinct
         .collect
       words.zipWithIndex.toMap
     }
 
-    val distinctDocument: Array[String] = docWordCount.map({ case SpotLDAInput(doc, word, count) => doc }).distinct.collect
+    val documentDictionary: DataFrame = docWordCountCache
+      .map({ case SpotLDAInput(doc, word, count) => doc })
+      .distinct
+      .zipWithIndex
+      .toDF(DocumentName, DocumentId)
+      .cache
 
-    // Create document Map Index, Document for further usage
-    val documentDictionary: Map[Int, String] = {
-      distinctDocument
-        .zipWithIndex
-        .sortBy(_._2)
-        .map(kv => (kv._2, kv._1))
-        .toMap
-    }
-    val docStrToID: Map[String, Int] = documentDictionary.map(_.swap)
+    // Forcing an action to cache results
+    documentDictionary.count()
 
     //Structure corpus so that the index is the docID, values are the vectors of word occurrences in that doc
-    val ldaCorpus: RDD[(Long, Vector)] = formatSparkLDAInput(docWordCount, docStrToID, wordDictionary)
+    val ldaCorpus: RDD[(Long, Vector)] =
+      formatSparkLDAInput(docWordCountCache,
+      documentDictionary,
+      wordDictionary,
+      sqlContext)
+
+    docWordCountCache.unpersist()
 
     //Instantiate optimizer based on input
     val optimizer = ldaOptimizer match {
       case "em" => new EMLDAOptimizer
-      case "online" => new OnlineLDAOptimizer().setMiniBatchFraction(0.05 + 1.0 / distinctDocument.size)
+      case "online" => new OnlineLDAOptimizer().setMiniBatchFraction(0.05 + 1.0 / documentDictionary.count)
       case _ => throw new IllegalArgumentException(s"Only em and online are supported but got $ldaOptimizer")
     }
 
-    logger.info("Running Spark LDA with params alpha = " + ldaAlpha + " beta = " + ldaBeta + " Max iters = " + maxIterations + " Optimizer = " + ldaOptimizer)
+    logger.info(s"Running Spark LDA with params alpha = $ldaAlpha beta = $ldaBeta " +
+      s"Max iterations = $maxIterations Optimizer = $ldaOptimizer")
     //Set LDA params from input args
 
     val lda =
-      new LDA().setK(topicCount).setMaxIterations(maxIterations).setAlpha(ldaAlpha).setBeta(ldaBeta).setOptimizer(optimizer)
+      new LDA()

```

Comment:
you really like to put each method on a separate line.... not sure which is better... single line is more concise but multiline calls out each step... is there a community standard on this?